packages/terra-application/src/page/PageHeaderButton.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+
+import styles from './PageHeaderButton.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The text for the button's aria-label attribute.
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * The icon to display for the associated action.
+   */
+  icon: PropTypes.element.isRequired,
+  /**
+   * Callback function for the selection of the button. If no function is
+   * provided, the button will render as disabled.
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Ref callback for button.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * Identifier for use during tests
+   */
+  testId: PropTypes.string,
```

Comment:
testId? Should we make this private or just be a regular id?

Reply:
I don't think this is being used at the moment, so I'll remove. Should be able to query in tests based on role/display just fine.